src/commands/logs.ts

```diff
@@ -0,0 +1,323 @@
+                            .setRequired(true)
+                    ),
+            { idHints: [] }
+        );
+    }
+
+    private async renderLogPage(
+        selectedCategory: Logs,
+        index: number,
+        userLog: ModerationLog,
+        cache: CachedPages
+    ) {
+        switch (selectedCategory) {
+            case Logs.User:
+                return this.userPage(userLog, cache);
+            case Logs.Warnings:
+                return this.warningsPage(index, userLog, cache);
+            case Logs.Timeouts:
+                return this.timeoutsPage(index, userLog, cache);
+            case Logs.Bans:
+                return this.bansPage(index, userLog, cache);
+            default:
+                return '';
+        }
+    }
+
+    private async getLogLength(
+        selectedCategory: Logs,
+        userLog: ModerationLog,
+    ) {
+        switch (selectedCategory) {
+            case Logs.Warnings:
+                return userLog.warnings?.length || 1;
+            case Logs.Timeouts:
+                return userLog.timeouts?.length || 1;
+            case Logs.Bans:
+                return userLog.bans?.length || 1;
+            default:
+                return 1;
+        }
+    }
+
+    private async userPage(
+        userLog: ModerationLog,
+        cache: CachedPages
+    ) {
+        if (cache.user.length) {
+            return cache.user[0];
+        }
+        const page = [
+            `**Warnings:** ${userLog.warnings?.length || 0}`,
+            `**Timeouts:** ${userLog.timeouts?.length || 0}`,
+            `**Bans:** ${userLog.bans?.length || 0}`,
+        ].join('\n');
+
+        cache.user.push(page);
+        return page;
+
+    }
+
+    private async warningsPage(
+        index: number,
+        userLog: ModerationLog,
+        cache: CachedPages
+    ) {
+        if (index < cache.warnings.length) {
+            return cache.warnings[index];
+        }
+        let page = ``;
+        if (!userLog.warnings) {
+            page = `User has no warnings`;
+        } else {
+            page = [
+                bold(`Warning ${index + 1}\n`),
+                bold('Moderator:'),
+                `<@${userLog.warnings[index].user}>`,
+                bold('Date'),
+                userLog.warnings[index].date,
+                bold('Reason'),
+                userLog.warnings[index].reason,
+            ].join('\n');
+        }
+
+        cache.warnings.push(page);
+        return page;
+    }
+
+    private async timeoutsPage(
+        index: number,
+        userLog: ModerationLog,
+        cache: CachedPages
+    ) {
+        if (index < cache.timeouts.length) {
+            return cache.timeouts[index];
+        }
+        let page = ``;
+        if (!userLog.timeouts) {
+            page = `User has no timeouts`;
+        } else {
+            page = [
+                bold(`Timeout ${index + 1}\n`),
+                bold('Moderator:'),
+                `<@${userLog.timeouts[index].user}>`,
```

Comment:
Could use the `userMention` helper from `discord.js` (here and the other 2 places we use this pattern in this file).
```suggestion
                userMention(userLog.timeouts[index].user),
```